From fbc1b80c32afe770f8806b4e6e52120ac1996e7f Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 18 Mar 2021 20:16:52 +0800
Subject: [PATCH 201/204] dt-bindings: iio: magnetometer: add DT binding for
 Voltafield AF8133J

Voltafield AF8133J is a quite simple magnetometer produced by Voltafield
Technology Corp, with dual power supplies (one for core and one for IO)
and a active-low reset pin, but without interrupts.

Add a device tree binding for it.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 .../iio/magnetometer/voltafield,af8133j.yaml  | 58 +++++++++++++++++++
 1 file changed, 58 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/iio/magnetometer/voltafield,af8133j.yaml

diff --git a/Documentation/devicetree/bindings/iio/magnetometer/voltafield,af8133j.yaml b/Documentation/devicetree/bindings/iio/magnetometer/voltafield,af8133j.yaml
new file mode 100644
index 000000000000..924a3c1d7c0e
--- /dev/null
+++ b/Documentation/devicetree/bindings/iio/magnetometer/voltafield,af8133j.yaml
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/iio/magnetometer/voltafield,af8133j.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Voltafield AF8133J magnetometer sensor
+
+maintainers:
+  - Icenowy Zheng <icenowy@aosc.io>
+
+properties:
+  compatible:
+    - voltafield,af8133j
+
+  reg:
+    maxItems: 1
+
+  reset-gpios:
+    description: |
+      an pin needed for AF8133J to set the reset state. This should be usually
+      active low.
+
+  avdd-supply:
+    description: |
+      an optional regulator that needs to be on to provide AVDD power (Working
+      power, usually 3.3V) to the sensor.
+
+  avdd-supply:
+    description: |
+      an optional regulator that needs to be on to provide DVDD power (Digital
+      IO power, 1.8V~AVDD) to the sensor.
+
+  mount-matrix:
+    description: an optional 3x3 mounting rotation matrix.
+
+required:
+  - compatible
+  - reg
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/interrupt-controller/irq.h>
+    #include <dt-bindings/gpio/gpio.h>
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        magnetometer@1c {
+            compatible = "voltafield,af8133j";
+            reg = <0x1c>;
+            avdd-supply = <&reg_dldo1>;
+            dvdd-supply = <&reg_dldo1>;
+            reset-gpios = <&pio 1 1 GPIO_ACTIVE_LOW>;
+        };
+    };
-- 
2.20.1

